homeassistant/components/smlight/sensor.py

```diff
@@ -0,0 +1,166 @@
+"""Support for SLZB-06 sensors."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import logging
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory, UnitOfInformation, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util.dt import utcnow
+
+from .const import DOMAIN, SCAN_INTERVAL, SMLIGHT_SLZB_REBOOT_EVENT
+from .coordinator import SmData, SmDataUpdateCoordinator
+from .entity import SmEntity
+
+UPTIME_DEVIATION = 5  # seconds
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class SmSensorEntityDescription(SensorEntityDescription):
+    """Class describing SMLIGHT sensor entities."""
+
+    value_fn: Callable[[SmData], float | None] = lambda _: None
+
+
+SENSORS = [
+    SmSensorEntityDescription(
+        key="esp32_temperature",
+        translation_key="core_temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+        value_fn=lambda x: x.sensors.esp32_temp,
+    ),
+    SmSensorEntityDescription(
+        key="zb_temperature",
+        translation_key="zigbee_temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+        value_fn=lambda x: x.sensors.zb_temp,
+    ),
+    SmSensorEntityDescription(
+        key="core_uptime",
+        translation_key="core_uptime",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda x: x.sensors.uptime,
+    ),
+    SmSensorEntityDescription(
+        key="socket_uptime",
+        translation_key="socket_uptime",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda x: x.sensors.socket_uptime,
+    ),
+    SmSensorEntityDescription(
+        key="ram_usage",
+        translation_key="ram_usage",
+        device_class=SensorDeviceClass.DATA_SIZE,
+        native_unit_of_measurement=UnitOfInformation.KILOBYTES,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
```

Comment:
are you planning for non diagnostic sensors in the future? otherwise could move this to the description definitions

Reply:
1. I think all sensors will be diagnsotics will move